SCClassLibrary/Common/Control/ServerStatus.sc

```diff
@@ -225,9 +225,45 @@ ServerStatusWatcher {
 
 	// final actions needed to finish booting
 	prFinalizeBoot {
-		ServerBoot.run(server);
-		server.sync;
-		server.initTree;
+		// this needs to be forked so that ServerBoot and ServerTree will definitely run before
+		// notified is true.
+		fork({
+			ServerBoot.run(server);
+			server.sync;
+			server.initTree;
+
+			this.notified = true;
+			server.changed(\serverRunning);
+		}, AppClock)
+	}
+
+	// This method attempts to recover from a loss of client-server contact,
+	// which is a serious emergency in live shows. So it posts a lot of info
+	// on the recovered state, and possibly helpful next user actions.
+
+	prHandleLoginWhenAlreadyRegistered { |clientIDFromProcess|
+		"% - handling login request though already registered - \n".postf(server);
+
+		// by default, only reset clientID if changed, to leave allocators untouched:
+		if (clientIDFromProcess != server.clientID) {
+			// make sure we can set the clientID, and set it
+			notified = false;
+			server.clientID_(clientIDFromProcess);
+			"// This seems to be a login after a crash, or from a new server object,\n"
+			"// so you may want to release currently running synths by hand:\n".postln;
+			"%.defaultGroup.release;\n".postf(server.cs);
+			"// and you may want to redo server boot finalization by hand:\n".postln;
+			"%.statusWatcher.prFinalizeBoot;\n".postf(server.cs);
+		} {
+			// same clientID, so leave all server resources in the state they were in!
+			"// This seems to be a login after a loss of network contact - \n"
+			"// - reconnected with the same clientID as before, so probably all is well.\n".postln;
```

Comment:
does this really need to be posted? i think this is reaching a point where it's going to be regarded as noise by users

Reply:
well, it is trying to be helpful in a rare emergency scenario - you are in the middle of playing a show, network drops out, you try to reconnect, and likely happy to see anything informative posted. I doubt that anyone would find that noisy in that situation. 
But maybe that is just me.

Reply:
gotcha, that makes sense. I think it would be nice if the post message was consistent with other messages coming out of this class, though. No other messages that I've seen start with "//", it seems arbitrary.

Reply:
actually since it should be noticeable, how about making it begin with *** for bold posting?

Reply:
just tried it - that does make the post-crash hints pop out nicely in the post window!

Reply:
all right!